packages/core/src/credential/Credential.ts

```diff
@@ -126,41 +126,32 @@ export function makeSigningData(
  * Verifies if the credential hash matches the contents of it.
  *
  * @param input - The credential to check.
- * @returns Whether they match or not.
  */
-export function verifyRootHash(input: ICredential): boolean {
-  return input.rootHash === calculateRootHash(input)
+export function verifyRootHash(input: ICredential): void {
+  if (input.rootHash !== calculateRootHash(input))
+    throw new SDKErrors.RootHashUnverifiableError()
 }
 
 /**
- * Verifies the data of the [[Credential]] object; used to check that the data was not tampered with, by checking the data against hashes.
+ * Verifies the data of the [[Credential]] object; used to check that the data was not tampered with,
+ * by checking the data against hashes. Throws if invalid.
  *
  * @param input - The [[Credential]] for which to verify data.
- * @returns Whether the data is valid.
  */
-export function verifyDataIntegrity(input: ICredential): boolean {
+export function verifyDataIntegrity(input: ICredential): void {
   // check claim hash
-  if (!verifyRootHash(input)) {
-    throw new SDKErrors.RootHashUnverifiableError()
-  }
+  verifyRootHash(input)
 
   // verify properties against selective disclosure proof
-  const { errors, verified } = Claim.verifyDisclosedAttributes(input.claim, {
+  Claim.verifyDisclosedAttributes(input.claim, {
     nonces: input.claimNonceMap,
     hashes: input.claimHashes,
   })
-  // TODO: how do we want to deal with multiple errors during claim verification?
-  if (!verified)
-    throw errors.length > 0 ? errors[0] : new SDKErrors.ClaimUnverifiableError()
 
   // check legitimations
   input.legitimations.forEach((legitimation) => {
-    if (!verifyDataIntegrity(legitimation)) {
-      throw new SDKErrors.LegitimationsUnverifiableError()
-    }
+    verifyDataIntegrity(legitimation)
```

Comment:
Could it be `input.legitimations.forEach(verifyDataIntegrity)`?

Reply:
oh yes it could!